kie-server-parent/kie-server-remote/kie-server-client/src/test/java/org/kie/server/client/LoadBalancerClientTest.java

```diff
@@ -0,0 +1,224 @@
+/*
+ * Copyright 2016 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+package org.kie.server.client;
+
+import java.util.Arrays;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import com.github.tomakehurst.wiremock.WireMockServer;
+import org.junit.Before;
+import org.junit.Test;
+import org.kie.remote.common.rest.KieRemoteHttpRequestException;
+import org.kie.server.api.model.KieServerInfo;
+import org.kie.server.api.model.ServiceResponse;
+import org.kie.server.client.balancer.BalancerStrategy;
+import org.kie.server.client.balancer.LoadBalancer;
+import org.kie.server.client.impl.AbstractKieServicesClientImpl;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.*;
+import static org.junit.Assert.*;
+
+public class LoadBalancerClientTest {
+
+    private String mockServerBaseUri1;
+    private String mockServerBaseUri2;
+    private String mockServerBaseUri3;
+
+    private WireMockServer wireMockServer1;
+    private WireMockServer wireMockServer2;
+    private WireMockServer wireMockServer3;
+
+    private KieServicesConfiguration config;
+
+    @Before
+    public void startServers() {
+        int port1 = BaseKieServicesClientTest.findFreePort();
+        wireMockServer1 = new WireMockServer(port1);
+        wireMockServer1.stubFor(get(urlEqualTo("/"))
```

Comment:
Would it be possible to move mock creation to a new method? If I am not mistaken, all three mock instances are nearly the same.